pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -608,6 +613,146 @@ fn resolve_option_inner(
     }
 }
 
+fn resolve_result_inner(
+    original: syn::TypePath,
+) -> syn::Result<(syn::Type, Option<CompositeTypeMacro>)> {
+    let segments = &original.path;
+    let last = segments
+        .segments
+        .last()
+        .ok_or(syn::Error::new(original.span(), "Could not read last segment of path"))?;
+
+    // Get the path of our Result type, to handle crate::module::Result pattern
+    // TODO - more robust / idiomatic way of handling this?
+    let mut without_type_args = original.path.clone();
+    without_type_args.segments.last_mut().unwrap().arguments = syn::PathArguments::None;
+ 
+    let (ok_ty, err_ty) = { 
+        if let syn::PathArguments::AngleBracketed(path_arg) = &last.arguments { 
+            if let Some(first_ty) = path_arg.args.first() {
+                // Since `pub type Result<T> = std::error::Result<T, OurError>
+                // is a common pattern, 
+                // we should support single-argument Result<T> style.
+                if path_arg.args.len() == 1 {
+                    (first_ty.clone(), None)
+                }
```

Comment:
Does this break if some sicko uses `type Result<E> = Result<SomeType, E>;`?

It's fine if it does, I'm just curious.

Reply:
It most likely would. I've never seen that pattern in the wild but it'd be nice if that could be accounted for - unfortunately I'm not sure if there's a good way to programmatically extract what type is used for the `Ok(T)` variant vs what type is used for the `Err(E)` variant at the level of a procedural macro.

Reply:
I don't think there's a realistic way since we aren't guaranteed to see the `type Whatever<?> = Result<T, E>;` decl.

Reply:
We would have to demand it is annotated by a macro.